conans/test/integration/toolchains/cmake/test_cmaketoolchain.py

```diff
@@ -719,3 +720,42 @@ def layout(self):
     presets = json.loads(client.load("build/14/generators/CMakePresets.json"))
     assert "architecture" in presets["configurePresets"][0]
     assert "toolset" not in presets["configurePresets"][0]
+
+
+def test_user_presets_custom_location():
+    client = TestClient()
+    conanfile = textwrap.dedent("""
+                import os
+                from conan import ConanFile
+                from conan.tools.cmake import cmake_layout, CMakeToolchain
+
+                class Conan(ConanFile):
+                    settings = "os", "arch", "compiler", "build_type"
+
+                    def generate(self):
+                        t = CMakeToolchain(self)
+                        t.user_presets_path = os.path.join(self.source_folder, "subproject")
```

Comment:
It would be nice to allow relative path here too ``t.user_presets_path = "subproject"``